app/domain/authentication/authn_jwt/authenticator.rb

```diff
@@ -0,0 +1,47 @@
+require 'command_class'
+
+module Authentication
+  module AuthnJwt
+    # Generic JWT authenticator that receive JWT vendor configuration and uses to validate that the authentication
+    # request is valid, and return conjur authn token accordingly
+    JwtAuthenticate ||= CommandClass.new(
+      dependencies: {
+        token_factory: TokenFactory.new
+      },
+      inputs: %i[jwt_configuration account username]
+    ) do
+      def call
+        validate_and_decode
+        get_identity
+        validate_restrictions
+        get_token
+      end
+
+      def get_identity
+        # Will be changed when real get identity implemented
+        unless @jwt_configuration.get_identity == "cucumber"
+          raise "wrong identity"
+        end
+      end
+
+      def validate_and_decode
+        unless @jwt_configuration.validate_and_decode
+          raise "validate and decode failed"
+        end
+      end
```

Comment:
can't we just call `@jwt_configuration.validate_and_decode` instead? won't `jwt_configuration` raise an error in case of a failure? this way you are masking the error.

Reply:
also - i would call it `validate_and_decode_token` to be more explicit

Reply:
Renamed to validate_and_decode_token
Currently this unless is placeholder because the configuration is dummy and just returning true. When we will reallly impliment the validate_and_decode_token we will change this function too